src/DssSpell.t.sol

```diff
@@ -1240,4 +1439,344 @@ contract DssSpellTest is DssSpellTestBase {
+    }
+
+    function test_lsskySpkFarm_deploymentAndInitialization() public {
+        assertEq(StakingRewardsLike(REWARDS_LSSKY_SPK).stakingToken(), LSSKY, "before: Wrong staking token");
+        assertEq(StakingRewardsLike(REWARDS_LSSKY_SPK).rewardsToken(), SPK, "before: Wrong rewards token");
+        assertEq(StakingRewardsLike(REWARDS_LSSKY_SPK).rewardsDistribution(), address(0), "before: Wrong rewards distribution");
+        assertEq(StakingRewardsLike(REWARDS_LSSKY_SPK).owner(), MCD_PAUSE_PROXY, "before: Wrong owner");
+
+        assertEq(VestedRewardsDistributionLike(REWARDS_DIST_LSSKY_SPK).dssVest(), MCD_VEST_SPK_TREASURY, "before: Wrong DssVest");
+        assertEq(VestedRewardsDistributionLike(REWARDS_DIST_LSSKY_SPK).stakingRewards(), REWARDS_LSSKY_SPK, "before: Wrong StakingRewards");
+        assertEq(VestedRewardsDistributionLike(REWARDS_DIST_LSSKY_SPK).gem(), SPK, "before: Wrong gem token");
+        assertEq(VestedRewardsDistributionLike(REWARDS_DIST_LSSKY_SPK).wards(MCD_PAUSE_PROXY), 1, "before: PauseProxy not authorized");
+
+        assertEq(
+            DssVestLike(MCD_VEST_SPK_TREASURY).gem(),
+            address(StakingRewardsLike(REWARDS_LSSKY_SPK).rewardsToken()),
+            "DssVest gem != StakingRewards rewardsToken"
+        );
+
+        assertEq(VestedRewardsDistributionLike(REWARDS_DIST_LSSKY_SPK).vestId(), 0, "before: vestId already set");
+        assertEq(VestedRewardsDistributionLike(REWARDS_DIST_LSSKY_SPK).lastDistributedAt(), 0, "before: Should not have distributed yet");
+        assertEq(StakingRewardsLike(REWARDS_LSSKY_SPK).totalSupply(), 0, "before: Should have no staked tokens");
+        assertEq(StakingRewardsLike(REWARDS_LSSKY_SPK).rewardRate(), 0, "before: Should have no reward rate");
+
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done(), "TestError/spell-not-done");
+
+        assertEq(StakingRewardsLike(REWARDS_LSSKY_SPK).rewardsDistribution(), REWARDS_DIST_LSSKY_SPK, "after: Wrong rewards distribution");
+
+        uint256 farmStatus = LockstakeEngineLike(LOCKSTAKE_ENGINE).farms(REWARDS_LSSKY_SPK);
+        assertEq(farmStatus, 1, "after: Farm not active in Lockstake Engine");
+
+        uint256 lsskySpkVestId = VestedRewardsDistributionLike(REWARDS_DIST_LSSKY_SPK).vestId();
+        assertGt(lsskySpkVestId, 0, "after: LSSKY->SPK vest stream not created");
+
+        (address lsskySpkUsr, , , , , , uint128 lsskySpkTot) =
+            DssVestLike(MCD_VEST_SPK_TREASURY).awards(lsskySpkVestId);
+        assertEq(lsskySpkUsr, REWARDS_DIST_LSSKY_SPK, "after: Wrong LSSKY->SPK vest recipient");
+        assertEq(lsskySpkTot, 975_000_000 * WAD, "after: Wrong LSSKY->SPK vest total");
+
+        assertGt(StakingRewardsLike(REWARDS_LSSKY_SPK).rewardRate(), 0, "after: LSSKY->SPK farm reward rate is zero");
+        assertGt(VestedRewardsDistributionLike(REWARDS_DIST_LSSKY_SPK).lastDistributedAt(), 0, "after: Should have distributed");
```

Comment:
Shouldn't lastDistributedAt be equal to block.timestamp? If so, it would be a more precise test.

Reply:
Fixed in [a803cf4](https://github.com/sky-ecosystem/spells-mainnet/pull/479/commits/a803cf4b757d39363fa93e46967fc92311ecc94f)